client/src/pages/EpisodeShowPage/index.tsx

```diff
@@ -382,7 +399,17 @@ export class WorkspaceView extends React.Component<any, any> {
       workspace.blocks.concat(workspace.childWorkspaces.map(cw => cw.blocks)),
     );
 
-    const reduxBlocks = props.reduxState.blocks.blocks.map(b => ({
+    // We want to snapshot the Redux state of this workspace
+    // The Redux state will have captured any changes made to the blocks
+    // That what the next three consts are doing (until the console.log)
+
+    const reduxBlocks = props.reduxState.blocks.blocks;
+
+    const reduxBlocksAssociatedWithWorkspace = reduxBlocks.filter(b =>
+      workspaceBlocks.find((b2: any) => b.id === b2.id),
+    );
+
```

Comment:
Nice, hopefully we'll see a little performance boost from filtering out the non-relevant blocks